### ISSUE TITLE
Adding metadata to audio files doesnt work

### ISSUE THREAD

**Issue by Tomhoogstra** [rg3/youtube-dl#11047]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.10.26*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x]  I've **verified** and **I assure** that I'm running youtube-dl **2016.10.26**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [x] Feature request (request for a new functionality)
- [x] Question
- [ ] Other

---

### Description of your *issue*, suggested solution and other information

Attempting to load extracted wav files into a java AudioInputStream but receiving an error due to the Bit rate and length properties not being present in the file.

A file converted on convert2mp3.net added the following properties to the file and it worked flawlessly
![Image of working file](http://puu.sh/rXDyD.png)
However, using youtube-dl, it retrieves a file with the following properties
![Image of broken file](http://puu.sh/rXDII.png)

Is there an argument to add this kind of information to a downloaded video? I tried a view such as meta data.
If there isn't would it be possible to add?
Thanks, keep up the good work.

Edit: Done some more digging, --add-metadata tag mentions that is adds the metadata for videos, but I tried it for audio anyway with no success. Decided to check the output when running it, and apparently using the following settings
"youtube-dl.py https://www.youtube.com/watch?v=6Cp6mKbRTQY --add-metadata --extract-audio --audio-format wav --output /Songs/6Cp6mKbRTQY.wav", the following error pops up at the very end.

![Error](http://puu.sh/rXF53.png)

**Comment by yan12125** [comment]
Could you paste the whole output of this command:
```
youtube-dl.py -v https://www.youtube.com/watch?v=6Cp6mKbRTQY --add-metadata --extract-audio --audio-format wav --output /Songs/6Cp6mKbRTQY.wav
```
Please paste as texts. Don't paste screenshots.

**Comment by Tomhoogstra** [comment]
http://pastebin.com/S2izHcVr

**Comment by yan12125** [comment]
@Tomhoogstra Could you try this:
```
youtube-dl.py -v https://www.youtube.com/watch?v=6Cp6mKbRTQY --add-metadata --extract-audio --audio-format wav --output /Songs/6Cp6mKbRTQY.%(ext)s
```

**Comment by Tomhoogstra** [comment]
@yan12125 
http://pastebin.com/RAchZ227

It creates a file with unknown extension
[http://puu.sh/s0Egf.png]

**Comment by yan12125** [comment]
The percentage sign in ```%(ext)s``` disappears from your command. If things are still broken after fixing the command, please test this: ```--output 6Cp6mKbRTQY.%%(ext)s```.

**Comment by Tomhoogstra** [comment]
@yan12125 Yes, using this command
youtube-dl.py https://www.youtube.com/watch?v=EbJtYqBYCV8 --audio-format wav --extract-audio --add-metadata --output /Songs/EbJtYqBYCV8.%%(ext)s

It adds the meta deta correctly, and even converts it to wav like I wanted it to.
So in the end it was a user error I guess.

**Comment by yan12125** [comment]
Yeah. Windows is tricky.

Closing as problem solved.

**Comment by Spenhouet** [comment]
This doesn't work for me.

`youtube-dl -x --audio-format "mp3" --audio-quality 0 -o "/Music/%(uploader)s/%(title)s.%(ext)s" --metadata-from-title "%(artist)s - %(title)s" --add-metadata https://www.youtube.com/watch?v=0sjR0-EG3BM`

The artist and other metadata is not set.
On Windows.

**Comment by yan12125** [comment]
@Spenhouet That's a known bug. See #5594

**Comment by Spenhouet** [comment]
@yan12125 I did find out what's the problem: If the first downloaded file format is `.webm` then, this file doesn't contain any metadata. Therefor the converted `.mp3` file also doesn't contain any metadata.

I added this command: `--format "bestaudio[ext=m4a]"`
Now it's downloading a `.m4a` file and saving the metadata correct (also in the converted `.mp3` file).

Could be related to this: https://github.com/rg3/youtube-dl/pull/9730
